vm/array.go

```diff
@@ -90,533 +81,547 @@ func (a *ArrayObject) shift() Object {
 	return value
 }
 
-// initArrayObject returns an array that contains given objects
-func initArrayObject(elements []Object) *ArrayObject {
-	return &ArrayObject{Elements: elements, Class: arrayClass}
+func (vm *VM) initArrayObject(elements []Object) *ArrayObject {
```

Comment:
Now we need vm to initialize built in type's objects, because their classes are stored in vm.